plugins/catalog-import/config.d.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface Config {
+  /**
+   * Configuration options for the catalog plugin.
+   */
+  catalog?: {
+    /**
+     * Catalog entity descriptor filename, defaults to "catalog-info.yaml"
+     * @visibility frontend
+     */
+    filename?: string;
+    /**
+     * A branch name used in pull request when registering existing component via UI
+     * Valid git refname required, see: https://git-scm.com/docs/git-check-ref-format
+     * Defaults to "backstage-integration"
+     * @visibility frontend
+     */
+    branchName?: string;
```

Comment:
Hm a little bit less sure about this one. First instinct is that this could just be a prop on the frontend components instead because it's very unlikely that anybody except one single place would want to consume it. Either way, it may make sense to name it `defaultGeneratedEntityBranchName` or similar, if it should stay at all.

Reply:
Settled on `pullRequestBranchName` for now